src/main/java/com/fasterxml/jackson/dataformat/xml/deser/FromXmlParser.java

```diff
@@ -932,12 +932,11 @@ public String nextTextValue() throws IOException
         switch (token) {
         case XmlTokenStream.XML_END_ELEMENT:
             if (_mayBeLeaf) {
-                // NOTE: this is different from nextToken() -- produce "", NOT null
                 _mayBeLeaf = false;
-                _currToken = JsonToken.VALUE_STRING;
+                _currToken = JsonToken.VALUE_NULL;
```

Comment:
Could you also add one-line comment referencing

    / / [dataformat-xml#584]: retain null values

Reply:
Actually I can do that.